packages/core/src/payment-integration/default-payment-integration-service.ts

```diff
@@ -192,4 +194,12 @@ export default class DefaultPaymentIntegrationService implements PaymentIntegrat
 
         return this._storeProjection.getState();
     }
+
+    async verifyCheckoutSpamProtection(): Promise<PaymentIntegrationSelectors> {
+        await this._store.dispatch(
```

Comment:
Maybe check if `shouldExecuteSpamCheck` is `true` first before trigger the action, similar to the logic we have here: https://github.com/bigcommerce/checkout-sdk-js/blob/605b50164333e6d37065ecd63f15a625a34e3155/packages/core/src/payment/payment-strategy-action-creator.ts#L55-L56